challenges/edabit-arrays.js

```diff
@@ -4,23 +4,135 @@
+     positiveNumbers.push(nums[i]);
+    	}
+    }
+  positiveNumbersCount = positiveNumbers.length;
+  positiveNumbersCountArray = [];
+  positiveNumbersCountArray.push(positiveNumbersCount);
+
+
+    for (var j = 0; j < nums.length; j++) {
+    if (nums[j] < 0){
+     negativeTotal += nums[j];
+    }
+  }
+  negativeNumbers.push(negativeTotal);
+
+    var positiveNegativeNumbers = positiveNumbersCountArray.concat(negativeNumbers);
+		return positiveNegativeNumbers;
+}
+
+
+countPosSumNeg([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, -11, -12, -13, -14, -15]);
+countPosSumNeg([92, 6, 73, -77, 81, -90, 99, 8, -85, 34]);
+countPosSumNeg([91, -4, 80, -73, -28])
+countPosSumNeg([]);
+
+
+
+
 // 2. Create a function that accepts an array and returns the last item in the array. The array can contain any of JavaScript's five primitive data types.
 // https://edabit.com/challenge/7JBTN4TbaxJQMdX9W
 
+function getLastItem(arr) {
+ return arr[arr.length-1]
+}
+
 // 3. Create a function that takes two arrays and combines them by alternatingly taking elements from each array in turn.
 // https://edabit.com/challenge/kaNYPQQWHSX4zGNhj
+function mergeArrays(a, b) {
+    var mergedArray = [];
+		var len = Math.max(a.length, b.length)
+
+    for (var i = 0; i < len; i++) {
+      if (i < a.length) {
+        mergedArray.push(a[i])
+      }
+      if (i < b.length) {
+        mergedArray.push(b[i])
+      }
+    }
+  return mergedArray;
 
+}
 
 // 4. Create a function that takes an array of numbers and remove the smallest value. If there are multiple elements with the same value, remove the one with a lower index. If you get an empty array, return an empty array.
 // https://edabit.com/challenge/9ukwiKyv8R9NHSt3d
+function removeSmallest(arr) {
+  var newArray = arr[0];
+	arr.forEach(function(number) {
+    if (number < newArray){
+    	newArray = number;
+    }
+  });
+  arr.splice(arr.indexOf(newArray), 1);
+
+  return arr;
+}
+
 
 
 // 5. Create a function that takes an array of numbers and returns the sum of the two lowest positive integers. No floats or empty arrays will be used in any of the test cases.
 // https://edabit.com/challenge/GNgCfKHWfQwByBNqa
 
+
+function sumTwoSmallestNums(array) {
+  var checkedArray = []
+  for (var i = 0; i < array.length; i++){
+    if (array[i] >= 0) {
+      checkedArray.push(array[i])
+    }
+  }
+
+    var firstLowestNumberIndex = checkedArray.indexOf(Math.min.apply(null, checkedArray))
+    var firstLowestNumber = checkedArray.splice(firstLowestNumberIndex, 1)
+    console.log('firstLowest number', firstLowestNumber)
+
+
+
+    var secondLowestNumberIndex = checkedArray.indexOf(Math.min.apply(null, checkedArray))
+    var secondLowestNumber = checkedArray.splice(secondLowestNumberIndex, 1)
+    console.log('secondLowestNumber', secondLowestNumber)
+
+    let lowestNumbersArray = firstLowestNumber.concat(secondLowestNumber)
+    console.log(firstLowestNumber.concat(secondLowestNumber))
+
+    return sum(lowestNumbersArray)
+
```

Comment:
This is a long solution. Here is an alternative:
```
+function sumTwoSmallestNums(arr) {
 +  let newArray = arr.filter( num => num> 0 )
 +    .sort(  (a, b) => parseInt(a) > parseInt( b ) )
 +  return newArray[ 0 ] + newArray[ 1 ]
 +}
```
This first filters out all negative numbers, then sorts them in ascending order. There first two values of the remaining array are returned.